app/routines/add_email_to_user.rb

```diff
@@ -7,13 +7,28 @@ class AddEmailToUser
   protected
 
   def exec(email_address_text, user, options={})
-    email_address = EmailAddress.new(value: email_address_text)
-    email_address.user = user
-    email_address.verified = options[:already_verified] || false
-    email_address.save
+    # If no email address, nothing to do
+    return if email_address_text.blank?
 
-    transfer_errors_from(email_address, { scope: :email_address }, true)
+    # If the email address already exists and is attached to the user, nothing to do
+    email_address = user.email_addresses.where(value: email_address_text).first
+    return if email_address.try(:verified)
+
+    # If it is a brand new email address, make it and send confirmation
+    if email_address.nil?
+      email_address = EmailAddress.new(value: email_address_text)
+      email_address.user = user
+      email_address.verified = options[:already_verified] || false
+
+      email_address.save
+      transfer_errors_from(email_address, { scope: :email_address }, true)
+
+      run(SendContactInfoConfirmation, email_address)
```

Comment:
actually SendContactInfoConfirmation doesn't send if already verified.  But that means I should be able to simplify this a bit.